web/yo/app/scripts/controllers/drugs.js

```diff
@@ -31,24 +61,40 @@ angular.module('oncokbApp')
 
             $scope.saveDrugName = function (newDrugName, drug) {
                 if (hasSameName(newDrugName, drug.uuid)) {
-                    modalError("Sorry", "Same name exists.", false, drug);
+                    modalError("Sorry", "Same name exists.", true, false, drug.uuid);
                 } else {
                     if (!newDrugName)
                         newDrugName = drug.drugName;
                     firebaseConnector.set('Drugs/' + drug.uuid + '/drugName', newDrugName);
                 }
             };
+
+            $scope.removeDrug = function (drug) {
+                if ($scope.drugMap[drug.uuid]) {
+                    var genes = _.map($scope.drugMap[drug.uuid], (gene) => {
+                        return gene.geneName
+                    });
+                    modalError("Sorry", "Can't delete this therapy, because it is found in the following gene pages.", false, false, drug.uuid, genes);
+                }
+                else {
+                    modalError("Attention", drug.drugName + " will be deleted.", false, true, drug.uuid);
+                }
+            };
         }]
     )
-    .controller('ModalErrorCtrl', function ($scope, $modalInstance, data) {
+    .controller('ModalErrorCtrl', function ($scope, $modalInstance, data, firebaseConnector) {
         $scope.errorTitle = data.errorTitle;
         $scope.errorMessage = data.errorMessage;
+        $scope.sameName = data.sameName;
         $scope.deleteDrug = data.deleteDrug;
+        if(data.genes){
+            $scope.drugGenes = data.genes.join(', ');
+        }
         $scope.cancel = function () {
             $modalInstance.dismiss('canceled');
         };
         $scope.confirm = function () {
-            firebaseConnector.set('Drugs/' + data.drug.uuid, null);
+            firebaseConnector.remove('Drugs/' + data.drugUuid);
```

Comment:
I think you should create a function called `removeDrug(uuid)` in firebaseconnector.js. You should only pass drug uuid to that function. In this way, we can prevent people write wrong path and mess up database.